src/components/implementation/no_interface/new_boot/boot_deps.h

```diff
@@ -0,0 +1,147 @@
+#include <print.h>
+
+#undef assert
+#define assert(node) do { if (unlikely(!(node))) { debug_print("assert error in @ "); *((int *)0) = 0;} } while(0)
+
+#include <cos_alloc.h>
+#include <cobj_format.h>
+#include <cos_types.h>
+
+/*the booter uses this to keep track of each comp*/
+struct comp_cap_info {
+	struct cos_compinfo cos_compinfo;
+	vaddr_t addr_start;
+	vaddr_t vaddr_mapped_in_booter;
+	vaddr_t upcall_entry;
+};
+struct comp_cap_info comp_cap_info[MAX_NUM_SPDS+1];
```

Comment:
github isn't letting me comment on your ( @ryuxin ) comment above about `__bump_mem_expand_range`.  We can make it public.  My comment was that if we do so, we need to rename it to be consistent with the rest of the API. @phanikishoreg was making a similar point.